datacube_stats/main.py

```diff
@@ -82,6 +82,18 @@ def list_statistics(ctx, param, value):
     ctx.exit()
 
 
+def gather_tile_indexes(tile_index, tile_index_file):
+    if not tile_index and not tile_index_file:
```

Comment:
Would prefer `tile_index is None` rather than `not`, same for `tile_index_file`.

Why? `not 0` is True, so is `not ''`. Now you have a much larger set of valid inputs, when in reality we want `tile_index :: Maybe( (int,int) )` and `tile_index_file :: Maybe( ValidFilesystemPathICanRead )`